### PR TITLE
docs: add feature flow documentation

### PR DESCRIPTION
**Pull request by ab-sandy**

### CODE REVIEW

File: features-doc/HEADLESS_ACCOUNT.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+![headless account image](https://user-images.githubusercontent.com/66050845/144535127-9357a992-e30a-4f10-8272-816a495c380e.png)
+
+1. When user login with 3rd party account for the first time, after user login will got user data without email address then will redirect to Accelbyte Player Portal for upgrade headless account
+    - **GET /iam/v3/public/users/me** to check whether the account has email address and verified or not. If user has email and verified then it is not headless and continue to login.
+2. User input email and request verification code
+    - **POST /iam/v3/public/namespaces/{namespace}/users/me/code/request** to get verification code. The context should be upgradeHeadlessAccount.
```
Comment (argaditya-ab) [inline features-doc/HEADLESS_ACCOUNT.md]: please wrap `upgradeHeadlessAccount` with code tag

File: features-doc/LOGIN.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+![Login image](https://user-images.githubusercontent.com/66050845/144534901-7cbc2198-63c9-4b94-aa8f-32b5d5d0e9da.png)
+
+1. When user click login button client will call the **GET /iam/v3/oauth/authorize** endpoint on IAM, the endpoint will generate request_id for this request and redirect the app to AB's login website serving login form
+    - Login website URL format:  https://{baseUrl}/auth/login?client_id={clientId}&redirect_uri={redirectUri}&request_id={requestId}
+    - URL example: https://demo.accelbyte.io/auth/login?client_id=8925bdce941c499b8817a8d1e2feab50&redirect_uri=https%3A%2F%2Fdemo.accelbyte.io&request_id=50f5839674004017878733ac742196d5    
+2. User input and submit login credential in login website
+3. Login website will call **POST /iam/v3/authenticate endpoint**, and if the user credentials are valid, IAM will redirect the user back to the Redirect_URI of the Client.
+    - Redirection URL format: {clientRedirectUri}/?code={code}&state={state}
```
Comment (argaditya-ab) [inline features-doc/LOGIN.md]: please wrap `{clientRedirectUri}/?code={code}&state={state}` with code tag

**Review by argaditya-ab — APPROVED** [review]

File: widget/src/index.tsx [hunk 3]
```diff
@@ -31,34 +31,42 @@ export const ElementSelector = {
   PurchaseButtonClassName: "ab-wpr-purchase-btn"
 };
 
-ReactDOM.render(
-  <Provider inject={[appState, appState.state.userSession]}>
-    <AppRoot/>
-  </Provider>
-  , document.getElementById(ElementSelector.AppRootId));
+if (document.getElementById(ElementSelector.AppRootId)) {
```
Comment (THZs) [inline widget/src/index.tsx]: This element is mandatory to be exist, please also print an error message to the console if this element doesn't exist.

File: widget/src/index.tsx [hunk 4]
```diff
@@ -31,34 +31,42 @@ export const ElementSelector = {
...
+    </Provider>
+    , document.getElementById(ElementSelector.AppRootId));
+}
 
-ReactDOM.render(
-  <Provider inject={[appState, appState.state.userSession]}>
-    <LoginButton/>
-  </Provider>
-  , document.getElementById(ElementSelector.AuthButtonRootId));
+if (document.getElementById(ElementSelector.AuthButtonRootId)) {
```
Comment (THZs) [inline widget/src/index.tsx]: How if we make it like this?
```
export const ElementSelector = {
  AppRootId: document.getElementById("ab-wpr-root"),
  AuthButtonRootId: document.getElementById("ab-wpr-login-button-root"),
  AuthButtonRootMobileId: document.getElementById("ab-wpr-login-button-root-mobile"),
  FooterPrivacyLinkId: document.getElementById("ab-wpr-footer-privacy-link"),
  PurchaseButtonClassName: document.getElementById("ab-wpr-purchase-btn")
};

if (ElementSelector.AuthButtonRootId) {
  ReactDOM.render(
    <Provider inject={[appState, appState.state.userSession]}>
      <LoginButton/>
    </Provider>
    , ElementSelector.AuthButtonRootId);
}
```

**Review by THZs — CHANGES_REQUESTED** [review]

**Review by THZs — APPROVED** [review]

**Pull request merged**